app/views/works/_edit_javascript.html.erb

```diff
@@ -81,30 +81,122 @@
+        value = types[i].value;
+        selected = related_identifier_type == key ? "selected" : "";
+        html += `<option value="${key}" ${selected}>${value}</option>`;
+      }
+      html += `</select>`;
+      return html;
+    }
+
+    var relationTypeSelectHtml = function(relationTypeId, relation_type) {
+      var i, html, related_identifier_type_options, key, value;
+      types = relationTypes();
+      html = `<select id="${relationTypeId}" name="${relationTypeId}">`;
+      for(i = 0; i < types.length; i++) {
+        key = types[i].key;
+        value = types[i].value;
+        selected = relation_type == key ? "selected" : "";
+        html += `<option value="${key}" ${selected}>${value}</option>`;
+      }
+      html += `</select>`;
+      return html;
+    }
+
+    var relationTypes = function() {
+      var types = [];
+      types.push({key: "IS_CITED_BY", value: "IsCitedBy"})
+      types.push({key: "CITES", value: "Cites"})
+      types.push({key: "IS_SUPPLEMENT_TO", value: "IsSupplementTo"})
+      types.push({key: "IS_SUPPLEMENTED_BY", value: "IsSupplementedBy"})
+      types.push({key: "IS_CONTINUED_BY", value: "IsContinuedBy"})
+      types.push({key: "CONTINUES", value: "Continues"})
+      types.push({key: "HAS_METADATA", value: "HasMetadata"})
+      types.push({key: "IS_METADATA_FOR", value: "IsMetadataFor"})
+      types.push({key: "IS_NEW_VERSION_OF", value: "IsNewVersionOf"})
+      types.push({key: "IS_PREVIOUS_VERSION_OF", value: "IsPreviousVersionOf"})
+      types.push({key: "HAS_VERSION", value: "HasVersion"})
+      types.push({key: "IS_VERSION_OF", value: "IsVersionOf"})
+      types.push({key: "IS_PART_OF", value: "IsPartOf"})
+      types.push({key: "HAS_PART", value: "HasPart"})
+      types.push({key: "IS_REFERENCED_BY", value: "IsReferencedBy"})
+      types.push({key: "REFERENCES", value: "References"})
+      types.push({key: "IS_DOCUMENTED_BY", value: "IsDocumentedBy"})
+      types.push({key: "DOCUMENTS", value: "Documents"})
+      types.push({key: "IS_COMPILED_BY", value: "IsCompiledBy"})
+      types.push({key: "COMPILES", value: "Compiles"})
+      types.push({key: "IS_VARIANT_FORM_OF", value: "IsVariantFormOf"})
+      types.push({key: "IS_ORIGINAL_FORM_OF", value: "IsOriginalFormOf"})
+      types.push({key: "IS_IDENTICAL_TO", value: "IsIdenticalTo"})
+      types.push({key: "IS_REVIEWED_BY", value: "IsReviewedBy"})
+      types.push({key: "REVIEWS", value: "Reviews"})
+      types.push({key: "IS_DERIVED_FROM", value: "IsDerivedFrom"})
+      types.push({key: "IS_SOURCE_OF", value: "IsSourceOf"})
+      types.push({key: "IS_OBSOLETED_BY", value: "IsObsoletedBy"})
+      types.push({key: "OBSOLETES", value: "Obsoletes"})
+      types.push({key: "IS_DESCRIBED_BY", value: "IsDescribedBy"})
+      types.push({key: "DESCRIBES", value: "Describes"})
+      types.push({key: "IS_REQUIRED_BY", value: "IsRequiredBy"})
+      types.push({key: "REQUIRES", value: "Requires"})
```

Comment:
Why does this need to be hardcoded? Can't `Datacite::Mapping::RelationType` be used here, too?